libraries/engage/a11y/components/LiveMessenger/index.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { UIEvents } from '../../../core';
+import LiveMessage from '../LiveMessage';
+import {
+  EVENT_LIVE_MESSAGE,
+  LIVE_MESSAGE_TYPE_ASSERTIVE,
+  LIVE_MESSAGE_TYPE_POLITE,
+} from './constants';
+
+/**
+ * The LiveMessenger component can be used to broadcast `aria-live` messages to assistive
+ * technology. It can be placed somewhere within the DOM, and receives its updates via
+ * EVENT_LIVE_MESSAGE event.
+ * @param {string} id An optional id for the messenger.
+ * @returns {JSX}
+ */
+const LiveMessenger = ({ id }) => {
+  const [assertive, setAssertive] = useState({
+    message: '', params: null,
+  });
+
+  const [polite, setPolite] = useState({
+    message: '', params: null,
+  });
+
+  const handleMessage = useMemo(() => (message, options) => {
```

Comment:
Is it on purpose, that the handler is not executed again, when the same event comes in more than once?
If not you will probably need  an actual callback rather than a memoized function wint an `undefined` return value (which you're not even interested in). `useMemo` is typically only used when you want to avoid multiple calls of the function when the parameters don't change.

In this case: When someone wants to handle a specific live message via sending an event, then another one and then the first one again, it will not handle the last event.

Reply:
Changed